server/src/main/java/org/elasticsearch/snapshots/SnapshotShardFailure.java

```diff
@@ -45,8 +45,13 @@
     private String nodeId;
     private ShardId shardId;
 
-    private SnapshotShardFailure() {
-
+    private SnapshotShardFailure(StreamInput in) throws IOException {
+        nodeId = in.readOptionalString();
+        shardId = new ShardId(in);
+        super.shardId = shardId.getId();
+        index = shardId.getIndexName();
+        reason = in.readString();
+        status = RestStatus.readFrom(in);
```

Comment:
guess this class could be a follow-up

Reply:
This is just a static method for reading the enum. It's not from Streamable.

Reply:
sure, I guess I just meant it from a design pattern perspective. anyways, let's ignore this :)